File: KotlinLibraryX/app-api/src/main/java/com/lee/api/adapter/ViewBindingAdapter.java
package com.lee.api.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.core.view.LayoutInflaterCompat;
import androidx.viewbinding.ViewBinding;

import com.lee.api.databinding.ItemContentBinding;
import com.lee.library.adapter.LeeViewAdapter;
import com.lee.library.adapter.LeeViewHolder;
import com.lee.library.adapter.listener.LeeViewItem;
import com.lee.library.adapter.manager.LeeViewItemManager;

import java.util.List;

/**
 * @author jv.lee
 * @date 2019/3/29
 * 封装的RecyclerViewAdapter
 */
public class ViewBindingAdapter<T> extends LeeViewAdapter<T> {

    /**
     * item类型管理器
     */
    private LeeViewItemManager<T> itemStyle;

    public ViewBindingAdapter(Context context, List<T> data) {
        super(context, data);
    }

    public ViewBindingAdapter(Context context, List<T> data, LeeViewItem<T> item) {
        super(context, data, item);
    }

    @NonNull
    @Override
    public LeeViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        //根据布局的类型 创建不同的ViewHolder
        LeeViewItem item = itemStyle.getLeeViewItem(viewType);
        int layoutId = item.getItemLayout();

        ViewBinding binding = ItemContentBinding.inflate(LayoutInflater.from(parent.getContext()), parent, false);
        LeeViewHolder viewHolder = new LeeViewHolder(binding.getRoot());

        //点击的监听
        if (item.openClick()) {
            setListener(viewHolder, item.openShake());
            //子view监听
            setChildListener(viewHolder, item.openShake());
        }
        return viewHolder;
    }
}

File: KotlinLibraryX/library/src/main/java/com/lee/library/adapter/binding/ViewBindingAdapter.kt
package com.lee.library.adapter.binding

import android.content.Context
import android.view.ViewGroup
import androidx.viewbinding.ViewBinding
import com.lee.library.adapter.base.BaseViewAdapter
import com.lee.library.adapter.base.BaseViewHolder

/**
 * [BaseViewAdapter] 的viewBinding实现 ,使用viewBinding解析view时可使用该适配器基类
 * @author jv.lee
 * @date 2019/3/29
 */
open class ViewBindingAdapter<T>(context: Context) : BaseViewAdapter<T>(context) {

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): BaseViewHolder {
        // 根据布局的类型 创建不同的ViewHolder
        val item = getItemStyles().getViewItem(viewType)
            ?: throw RuntimeException("itemStyle.getViewItem is null.")
        val viewBinding = item.getItemViewAny(parent.context, parent) as? ViewBinding
            ?: throw RuntimeException("itemStyle.getItemViewAny is null.")
        val viewHolder = ViewBindingHolder(viewBinding)

        if (item.openClick()) {
            // itemView点击事件设置监听
            setListener(viewHolder, item.openShake())
            // childView点击事件设置监听
            setChildListener(viewHolder, item.openShake())
        }
        return viewHolder
    }
}